### PR TITLE
[survey] remove dependency to website

### PR DESCRIPTION
**Pull request by rop-odoo**
- https://www.odoo.com/web?#id=23849&view_type=form&model=project.task&menu_id=5195

This is an improvement for the Odoo community users as they won't have to install `website` anymore to use `survey`.

### CODE REVIEW

File: addons/survey/controllers/main.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ def _check_bad_cases(self, survey, token=None):
             return werkzeug.utils.redirect("/survey/")
 
         # In case of auth required, block public user
-        if survey.auth_required and request.env.user == request.website.user_id:
+        if survey.auth_required and request.env.user == request.env.ref('base.public_user'):
```
Comment (tde-banana-odoo) [inline addons/survey/controllers/main.py]: Couldn't we use _is_public ?

File: addons/test_mail/tests/test_performance.py [hunk 2]
```diff
@@ -139,7 +139,7 @@ def test_adv_activity_full(self):
             'default_res_model': 'mail.test.activity',
         })
 
-        with self.assertQueryCount(admin=47, emp=52):  # com runbot 46 - 51 // test_mail only: 35 - 40
+        with self.assertQueryCount(admin=49, emp=54):  # com runbot 48 - 53 // test_mail only: 35 - 40
```
Comment (tde-banana-odoo) [inline addons/test_mail/tests/test_performance.py]: Could you investigate why this change is necessary ?

File: addons/survey/controllers/main.py [hunk 3]
```diff
@@ -74,13 +74,13 @@ def start_survey(self, survey, token=None, **post):
         # Manual surveying
         if not token:
             vals = {'survey_id': survey.id}
-            if request.website.user_id != request.env.user:
+            if request.env.ref('base.public_user') != request.env.user:
```
Comment (tde-banana-odoo) [inline addons/survey/controllers/main.py]: Couldn't we use _is_public ?

File: addons/survey/views/survey_templates.xml [hunk 4]
```diff
@@ -1,9 +1,42 @@
...
+        <script src="/web/static/lib/nvd3/nv.d3.js"></script>
+        <script type="text/javascript" src="/web/static/src/js/services/session.js"></script>
+        <script type="text/javascript" src="/survey/static/src/js/survey_result.js" />
+        <script type="text/javascript" src="/survey/static/src/js/survey.js" />
+        <script type="text/javascript" src="/survey/static/src/js/tour_test_survey.js"></script>
+        <script type="text/javascript" src="/web/static/src/js/fields/field_utils.js"></script>
+        <link href="/web/static/lib/nvd3/nv.d3.css" rel="stylesheet" type="text/css"></link>
+        <link rel="stylesheet" type="text/less" href="/web/static/src/less/import_bootstrap.less"/>
+        <link rel="stylesheet" type="text/less" href="/web/static/src/less/fonts.less" />
+        <!-- Some helper classes are (like mt- mb-) are required for survey templates  -->
```
Comment (mart-e) [inline addons/survey/views/survey_templates.xml]: twice "are"

File: addons/survey/views/survey_templates.xml [hunk 5]
```diff
@@ -1,9 +1,42 @@
...
+    </template>
+
+    <!-- Forbidden error messages-->
+    <template id="403">
+        <t t-call="survey.layout">
+            <div id="wrap">
+                <div class="container">
+                    <h1 class="mt32">403: Forbidden</h1>
+                    <p>The page you were looking for could not be authorized.</p>
+                    <p>Maybe you were looking for this popular page ?</p>
```
Comment (mart-e) [inline addons/survey/views/survey_templates.xml]: I don't think it is a popular page, change the message. Also the link is not called "Home".

File: addons/survey/views/survey_templates.xml [hunk 6]
```diff
@@ -68,9 +101,42 @@
...
+        <t t-call="web.layout">
+            <t t-set="head">
+                <t t-call-assets="web.assets_common"/>
+                <t t-call-assets="survey.survey_assets"/>
+            </t>
+            <body>
+                <header>
+                    <div><title>Odoo</title></div>
+                    <div class="text-center">
+                        <img t-attf-src="/web/binary/company_logo{{ '' }}"/>
```
Comment (mart-e) [inline addons/survey/views/survey_templates.xml]: remove the `{{ '' }}`
